libs/core/src/lib/checkbox/checkbox/checkbox.component.ts

```diff
@@ -144,6 +144,29 @@ export class CheckboxComponent implements ControlValueAccessor {
         this._changeDetectorRef.detectChanges();
     }
 
+    /** @hidden Updates checkbox state on mouse click
+     * and prevents from double check update from label-input binding */
+    public checkByClick(event: Event) {
+        this.nextValue();
+        event.preventDefault();
+    }
+
+    /** @hidden Updates checkbox state on spacebar key
+     * and prevents from double check update from label-input binding */
+    public checkByKey(event: KeyboardEvent): void {
+        if (event.keyCode === 32) {
```

Comment:
lets move key `32` to constant

Reply:
also move that if statement to function because it is called 2 times here and line 165 `is'x'Entered()`

Reply:
Moving conditional statement to separate function is a good idea, but I don't think there is a need to create variable for space bar key code.